sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamTest.scala

```diff
@@ -228,7 +228,8 @@ trait StreamTest extends QueryTest with SharedSQLContext with Timeouts {
    */
   def testStream(
       _stream: Dataset[_],
-      outputMode: OutputMode = OutputMode.Append)(actions: StreamAction*): Unit = {
+      outputMode: OutputMode = OutputMode.Append,
+      ignoreThreadDeathCause: Boolean = false)(actions: StreamAction*): Unit = {
```

Comment:
This `ignore` field is really hard to understand as a user of this function. And adding a field to just for one test does not make sense. Rather make an action ExpectStreamThreadException.